wcfapp/WcfServiceConsoleApp/Program.cs

```diff
@@ -0,0 +1,35 @@
+﻿using System;
+using System.ServiceModel;
+using System.Threading;
+
+namespace WcfServiceConsoleApp
+{
+    class Program
+    {
+        static void Main(string[] args)
+        {
+            // Create the ServiceHost
+            ServiceHost host = new ServiceHost(typeof(Service1));
+
+            try {
+                // Open the ServiceHost to start listening for messages
+                host.Open();
+
+                foreach (var uri in host.BaseAddresses)
+                {
+                    Console.WriteLine("The service is ready at {0}", uri);
+                }
+
+                // Leave the servicing running
+                Console.WriteLine("The service is running...");
+                Thread.Sleep(-1);
+            }
+            finally
+            {
+                host?.Close();
+                Console.WriteLine("The service is closed");
```

Comment:
host.Close can throw. I also don't see how this code can be reached.

Reply:
Yes, weird code after changing from `Console.ReadLine()` to `Sleep(-1)` 😄. Alright, I changed `finally` to `catch` so it can be reached at least... theoretically.